### PR TITLE
[ruff] Enable `SLF001` `private-member-access`

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Enable ruff `SLF001` (`private-member-access`). Replaces previously enabled pylint rule.

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12270** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12270" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12270?utm_source=stack-comment).

File: helm/dagster/schema/schema/utils/helm_template.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def render(
             k8s_objects = [k8s_object for k8s_object in yaml.full_load_all(templates) if k8s_object]
             if self.model:
                 k8s_objects = [
-                    self.api_client._ApiClient__deserialize_model(  # pylint: disable=W0212
+                    self.api_client._ApiClient__deserialize_model(  # pylint: disable=W0212  # noqa: SLF001
```
Comment (sryza) [inline helm/dagster/schema/schema/utils/helm_template.py:80]: Should we get rid of the pylint disable here? A bunch of other instances of this as well.

Comment (smackesey) [reply helm/dagster/schema/schema/utils/helm_template.py:80]: Behold upstack: https://github.com/dagster-io/dagster/pull/12162

**Review by alangenfeld — APPROVED** [review]

**Comment by smackesey** [comment]
@smackesey queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12270).

**Comment by smackesey** [comment]
[Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12270) failed to merge this pull request for an unknown reason. Try your merge again, or report a bug if you see this consistently.

**Comment by smackesey** [comment]
@smackesey queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12270).

**Comment by smackesey** [comment]
[Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12270) rebased this pull request as part of a merge.

**Comment by smackesey** [comment]
[Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12270) failed to merge this pull request for an unknown reason. Try your merge again, or report a bug if you see this consistently.

**Pull request merged**